### PR TITLE
ore: remove unused functions

### PR DESCRIPTION
**Pull request by petrosagg**
We don't use the `Lottery` or `MaybeFuture` abstractions anywhere so remove them. Also, `Either` Future extensions are now provided by `futures-rs` https://docs.rs/futures/0.3.17/futures/future/trait.FutureExt.html#method.left_future

The first and last use of `Lottery` was removed in `2019` in 6b1bd87ed8e0fb864e883f8507ad058916c4fe47 and `MaybeFuture` was used before we were purifying SQL statements.

### CODE REVIEW

**Comment by umanwizard** [comment]
Ore is Apache-licensed, so it’s possible someone external is using these.

On the other hand, we don’t publish it or advertise its open-sourceness.

File: src/ore/src/fmt.rs [hunk 1]
```diff
@@ -86,18 +86,6 @@ pub trait FormatBuffer: AsRef<[u8]> {
     /// Returns the number of bytes in the buffer.
     fn len(&self) -> usize;
 
-    /// Reports whether the buffer is empty.
-    fn is_empty(&self) -> bool {
```
Comment (antiguru) [inline src/ore/src/fmt.rs:90]: Keep `is_empty` to complement `len`.

File: src/ore/src/metrics/third_party_metric.rs [hunk 2]
```diff
@@ -82,23 +72,3 @@ impl<T: MetricVecBuilder> ThirdPartyMetric<MetricVec<T>> {
...
-
-impl<P: Atomic> ThirdPartyMetric<GenericGaugeVec<P>> {
-    /// Creates a delete-on-drop gauge that can be scraped by third parties.
-    pub fn get_third_party_delete_on_drop_gauge<'a, L: PromLabelsExt<'a>>(
-        &self,
-        labels: L,
-    ) -> DeleteOnDropGauge<'a, P, L> {
-        self.inner.get_delete_on_drop_gauge(labels)
-    }
-}
```
Comment (antiguru) [inline src/ore/src/metrics/third_party_metric.rs:104]: The two impls should be used at some point, better to keep I think.

Comment (petrosagg) [reply src/ore/src/metrics/third_party_metric.rs]: thanks for the review! re-added

**Review by benesch — APPROVED** [review]

**Pull request merged**